test/common/common_arena_constraints.h

```diff
@@ -153,6 +171,15 @@ class system_info {
 
     system_info() {
         hwloc_require_ex(hwloc_topology_init, &topology);
+#if __HWLOC_WINDOWS_AFFINITY_RESPECTING_INTERFACES_PRESENT
```

Comment:
Looks confusing. Why WINDOWS specific? Can you please add a comment in the code what this macros is about and why it needs to be used instead of `__HWLOC_RESTRICT_TO_CPUBINDING_TOPOLOGY_FLAG_PRESENT`?